commons/src/test/java/org/eclipse/kapua/commons/setting/KapuaSettingExceptionTest.java

```diff
@@ -20,72 +20,114 @@
-        assertNull("Expected and actual values should be the same!", kapuaSettingException.getCause());
-        assertEquals("Expected and actual values should be the same!", kapuaSettingException.getMessage(), "Error: " + null);
+        assertEquals("Expected and actual values should be the same!", "INVALID_RESOURCE_NAME", kapuaSettingException.getCode().toString());
+        assertNull("Null expected!", kapuaSettingException.getCause());
+        assertEquals("Expected and actual values should be the same!", "Error: " + null, kapuaSettingException.getMessage());
+        assertEquals("Expected and actual values should be the same!", kapuaErrorMessage, kapuaSettingException.getKapuaErrorMessagesBundle());
     }
 
     @Test
-    public void extendedConstructorWithThrowableAndObjectArgTest() {
+    public void kapuaSettingExceptionWithThrowableAndObjectArgTest() {
         Object[] arguments = new Object[]{"invalid_resource_name", 1, 10L, 10.34, 5.99f, 'D', "", true, false, -128, 127, -32768, 32767, -2147483648, 2147483647, -9223372036854775808L, 9223372036854775807L};
         for (Object value : arguments) {
             KapuaSettingException kapuaSettingException = new KapuaSettingException(KapuaSettingErrorCodes.INVALID_RESOURCE_NAME, throwable, value);
-            assertEquals("Expected and actual values should be the same!", kapuaSettingException.getCode().toString(), "INVALID_RESOURCE_NAME");
-            assertEquals("Expected and actual values should be the same!", kapuaSettingException.getCause(), throwable);
+            assertEquals("Expected and actual values should be the same!", "INVALID_RESOURCE_NAME", kapuaSettingException.getCode().toString());
+            assertEquals("Null expected!", throwable, kapuaSettingException.getCause());
             assertEquals("Expected and actual values should be the same!", kapuaSettingException.getMessage(), "Error: " + value);
+            assertEquals("Expected and actual values should be the same!", kapuaErrorMessage, kapuaSettingException.getKapuaErrorMessagesBundle());
+        }
+    }
+
+    @Test(expected = KapuaSettingException.class)
+    public void throwingKapuaSettingExceptionKapuaSettingErrorCodesTest() throws KapuaSettingException {
+        for (KapuaSettingErrorCodes code : kapuaSettingErrorCodes) {
+            throw new KapuaSettingException(code);
+        }
+    }
+
+    @Test(expected = KapuaSettingException.class)
+    public void throwingKapuaSettingExceptionKapuaSettingErrorCodesArgumentsTest() throws KapuaSettingException {
+        for (KapuaSettingErrorCodes code : kapuaSettingErrorCodes) {
+            throw new KapuaSettingException(code, "String", true, 1, 10L);
+        }
+    }
+
+    @Test(expected = KapuaSettingException.class)
+    public void throwingKapuaSettingExceptionKapuaSettingErrorCodesCauseArgumentsTest() throws KapuaSettingException {
```

Comment:
Please rename this test to: "throwingKapuaSettingErrorCodesWithCauseAndArguments"